python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -132,13 +143,29 @@ def __init__(self,
         for local_path in self.config["file_mounts"].values():
             assert os.path.exists(local_path)
 
+        global _metric_server_started
+        if not _metric_server_started:
+            try:
+                logger.info(
+                    "Starting autoscaler metrics server on port {}".format(
+                        AUTOSCALER_METRIC_PORT))
+                prometheus_client.start_http_server(
+                    AUTOSCALER_METRIC_PORT,
+                    registry=self.prom_metrics.registry)
+                _metric_server_started = True
+            except Exception as e:
+                logger.error(
+                    "An error occurred while starting the metrics server.")
```

Comment:
I would not re-raise here and let it just fail.

Also, you can use `logger.exception("msg")` to capture the stacktrace automagically.